HealthVault/Classes/Clients/MHVPersonClient.m

```diff
@@ -53,13 +56,70 @@ - (instancetype)initWithConnection:(id<MHVConnectionProtocol>)connection
 
 - (void)getApplicationSettingsWithCompletion:(void(^)(MHVApplicationSettings *_Nullable settings, NSError *_Nullable error))completion
 {
+    MHVASSERT_PARAMETER(completion);
+    
+    if (!completion)
+    {
+        return;
+    }
     
+    MHVMethod *method = [MHVMethod getApplicationSettings];
+    
+    [self.connection executeHttpServiceOperation:method
+                                      completion:^(MHVServiceResponse * _Nullable response, NSError * _Nullable error)
+     {
+         if (error)
+         {
+             completion(nil, error);
+             
+             return;
+         }
+         
+         MHVApplicationSettings *applicationSettings;
+         
+         XReader *reader = [[XReader alloc] initFromString:response.infoXml];
+         
+         if ([reader readStartElementWithName:@"info"])
+         {
+             applicationSettings = (MHVApplicationSettings *)[reader readElement:@"app-settings" asClass:[MHVApplicationSettings class]];
+         }
+         
+         if (!applicationSettings)
+         {
+             completion(nil, [NSError error:[NSError MHVUnknownError] withDescription:@"The GetApplicationSettings response is invalid."]);
+             
+             return;
+         }
+         
+         completion(applicationSettings, nil);
+     }];
 }
 
-- (void)setApplicationSettingsWithRequestParameters:(NSString *)requestParameters
-                                         completion:(void(^_Nullable)(NSError *_Nullable error))completion
+- (void)setApplicationSettings:(MHVApplicationSettings *)applicationSettings
+                    completion:(void(^_Nullable)(NSError *_Nullable error))completion
 {
+    MHVASSERT_PARAMETER(applicationSettings);
+
+    if (!applicationSettings)
+    {
+        if (completion)
+        {
+            completion([NSError MVHRequiredParameterIsNil]);
+        }
+        return;
+    }
     
+    MHVMethod *method = [MHVMethod setApplicationSettings];
+    method.parameters = [self bodyForSetApplicationSettings:applicationSettings];
```

Comment:
Because this must be valid XML, we should check if bodyForSetApplicationSettings: returns nil, and if so, provide the caller with a detailed error explaining to them that the string is not valid XML.